Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CORE_X12AuditDataTransform.java

```diff
@@ -379,89 +422,116 @@ private static boolean areRequiredRequestFieldsNull(Object msg) {
                 return true;
             } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getTimeStamp())) {
                 LOG.error("CORE X12 Batch TimeStamp is empty...");
+
+                return true;
+            }
+        } else if (msg instanceof COREEnvelopeRealTimeRequest) {
+            coreEnvelopeRealTimeRequest = (COREEnvelopeRealTimeRequest) msg;
```

Comment:
These are repetitive.  You could create a helper method that does something like areRequiredFieldsNull(String  ruleVersion, String payloadType, String processingMode ...., String messageType).  And just call that from each parent if statement.  messageType would be used for logging.